### PR TITLE
Add data/dataops fundamentals 

### PR DESCRIPTION
**Pull request by TessFerrandez**
This PR adds a section on how to apply engineering fundamentals to data projects and links to the https://github.com/Azure-Samples/ repo for samples.

The idea is to expand this later in a similar way to the ML fundamentals

### CODE REVIEW

File: data-fundamentals/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Data and DataOps fundamentals
+
+Most projects involve some type of data storage, data processing and data ops. For these projects, as with all projects, we follow the general guidelines laid out in other sections around security, testing, observability, CI/CD etc.
+
+## Goal
+
+This goal of this section is to help guide on some of the specifics around how to apply these fundamentals to data projects or data portions of a project.
```
Comment (tompaana) [inline data-fundamentals/README.md:7]: **The** goal...

The sentence does not skimp on words. How about: "The goal of this section is to briefly describe how to apply the fundamentals to data heavy projects or portions of the project."

File: data-fundamentals/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+- [End to End and Azure Technology Samples](#end-to-end-and-azure-technology-samples)
+
+## Data Tiering (Data Quality)
+
+Develop a common understanding of the quality of your datasets so that everyone understands the quality of the data, and expected use cases and limitations.
+
+A common data quality model is `Bronze`, `Silver`, `Gold`
+
+- **Bronze:** This is a landing area for your raw datasets with no to minimal data transformations applied, and therefore are optimized for writes / ingestion. Treat these datasets as an immutable, append only store.
+- **Silver:** These are cleansed, semi-processed datasets. These conform to a known schema and predefined data invariants and might have further data augmentation applied. These are typically used by Data Scientists.
```
Comment (tompaana) [inline data-fundamentals/README.md:27]: Nit: I don't think "data scientists" should be capitalized (despite how great they are!)

File: data-fundamentals/README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+- **Gold:** These are highly processed, highly read-optimized datasets primarily for consumption of business users. Typically, these are structured in your standard Fact and Dimension tables.
```
Comment (tompaana) [inline data-fundamentals/README.md:28]: Nit: "Fact", "Dimensions" capitalized - should they be really?

File: data-fundamentals/README.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+Divide your data lake into three major areas containing your Bronze, Silver and Gold datasets.
+
+> Note: additional storage areas for malformed data, intermediate (sandbox) data, and libraries/packages/binaries are also useful when designing your storage organization
```
Comment (tompaana) [inline data-fundamentals/README.md:32]: Nit: "additional" -> "Additional" and maybe end the sentence with punctuation?

I'm really going off the rails with my focus on capitalization, aren't I?

File: data-fundamentals/README.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+## Data Validation
+
+Validate data early in your pipeline.
+
+- Add data validation between the Bronze and Silver datasets. By validating early in your pipeline, you can ensure all succeeding datasets conform to a specific schema and known data invariants. This can also potentially prevent data pipeline failures in case of unexpected changes to the input data.
```
Comment (tompaana) [inline data-fundamentals/README.md:38]: Nit: I understand "succeeding" in the context, but it seems off. Could be removed completely or modified e.g. "...all datasets passing validation...".

File: data-fundamentals/README.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+- Data that does not pass this validation stage can be rerouted to a Malformed Record store for diagnostic purposes.
```
Comment (tompaana) [inline data-fundamentals/README.md:39]: How about: "...to a record store dedicated for malformed data..."?

File: data-fundamentals/README.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+- **Silver:** These are cleansed, semi-processed datasets. These conform to a known schema and predefined data invariants and might have further data augmentation applied. These are typically used by Data Scientists.
+- **Gold:** These are highly processed, highly read-optimized datasets primarily for consumption of business users. Typically, these are structured in your standard Fact and Dimension tables.
+
+Divide your data lake into three major areas containing your Bronze, Silver and Gold datasets.
+
+> Note: additional storage areas for malformed data, intermediate (sandbox) data, and libraries/packages/binaries are also useful when designing your storage organization
+
+## Data Validation
+
+Validate data early in your pipeline.
```
Comment (tompaana) [inline data-fundamentals/README.md:36]: Nit: These "rule of thumb" sentences under other headings don't have punctuation. The "rule of thumb" of consistency says they should be all the same. Uniqueness is sadly frowned upon here. Conformity is the way. Resistance is futile.

File: data-fundamentals/README.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+## CI/CD, Source Control and Code Reviews
+
+- All artifacts needed to build the data pipeline from scratch should be in source control. This included infrastructure-as-code artifacts, database objects (schema definitions, functions, stored procedures etc.), reference/application data, data pipeline definitions and data validation and transformation logic.
+- Any new artifacts (code) introduced to the repository should be code reviewed, both automatically (linting, credential scanning etc.) and peer reviewed.
+- There should be a safe, repeatable process (CI/CD) to move the changes through dev, test and finally production.
+
+## Security and Configuration
+
+- Maintain a central, secure location for sensitive configuration such as database connection strings that can be accessed by the appropriate services within the specific environment.
+- On Azure this is typically solved through securing secrets in a KeyVault per environment, then having the relevant services query KeyVault for the configuration
```
Comment (tompaana) [inline data-fundamentals/README.md:65]: Nit: KeyVault -> Key Vault

**Review by tompaana — CHANGES_REQUESTED** [review]
That's one neat, compact information package! Needs a wee bit polishing.

File: data-fundamentals/README.md (hunk 1 above)
Comment (TessFerrandez) [reply data-fundamentals/README.md]: fixed

File: data-fundamentals/README.md (hunk 2 above)
Comment (TessFerrandez) [reply data-fundamentals/README.md]: fixed

**Review by tompaana — APPROVED** [review]

**Review by devlace — APPROVED** [review]
LGTM 👍 Tomi was so thorough with his review 😊

**Pull request merged**